grammar.js

```diff
@@ -39,13 +39,15 @@ const primitive_types = numeric_types.concat(['bool', 'str', 'char'])
 module.exports = grammar({
   name: 'rust',
 
-  extras: $ => [/\s/, $.line_comment, $.block_comment],
+  extras: $ => [/\s/, $.line_doc_comment, $.line_comment, $.block_comment, $.block_doc_comment],
```

Comment:
I think it should be possible to expose doc contents without introducing new named nodes for doc comments by making the `doc_content` nodes optional fields on `line_comment` and `block_comment`. For example, line comments could be structured like this:

```
// Line comment
/// Doc comment

---

(source
  (line_comment)
  (line_comment
    docs: (doc_content)))
```

Then the query for injecting markdown only needs to match against the `doc_content` node and highlights queries won't need any changes.

Reply:
It's better to make these two separate nodes.  In this repository's query file we would just link `line_comment`/`block_comment` to `@comment` and then `line_doc_comment`/`block_doc_comment` to `@comment.doc`.

Anything marked as `@comment.doc` will automatically use `@comment` as a fallback so anything using the queries (like colors schemes) won't need any changes in the first place.

Reply:
You could still capture these with separate `comment` and `comment.doc` captures by matching on the presence or absence of the `docs` field:

```scm
(line_comment !docs) @comment
(line_comment docs: _) @comment.doc
; same for block comments
```

Reply:
![chart](https://user-images.githubusercontent.com/126196431/222838392-e77f98ad-71c1-4bd6-894e-13624b1932d5.png)